framework/CodeInterfaces/PHISICS/combine.py

```diff
@@ -0,0 +1,173 @@
+    with open(csvFile, 'r') as file:
+      for line in file:
+        if countLine == 0 and csvFile == self.phisicsCSV:
+          self.numOfParameters = self.getNumOfParameters(line)
+        countLine = countLine + 1
+        csvDict[countLine] = line
+    return csvDict
+  
+  def getNumOfParameters(self,line):
+    """
+      Returns the number of parameters that are on one line of the csv file.
+      @ In, line, string, csv file first line
+      @ Out, getNumOfParameters, integer
+    """
+    return len(line.split(','))
+
+  def getNumOfLines(self,csvFile):
+    """
+      Counts the number of lines in the PHISICS or RELAP csv.
+      @ In, csvFile, string, csv file name
+      @ Out, count, integer
+    """
+    count = 0
+    with open(csvFile, 'r') as file:
+      for line in file:
+        count = count + 1
+    return count
+  
+  def joinLine(self,phisicsList,relapList):
+    """
+      Joins the PHISICS and RELAP line into one list.
+      @ In, phisicsList, list, list contianing a csv PHISICS line
+      @ In, relapList, list, list containing a csv RELAP line
+      @ Out, joinedList, list, joined list of parameters
+    """
+    joinedList = []
+    joinedList.append(phisicsList.strip('\n').strip('\r'))
+    joinedList.append(relapList.strip('\n').strip('\r'))
+    return joinedList
+  
+  def joinCSV(self,paramDict):
+    """
+      Joins the RELAP csv and PHISICS csv based on the time lines selected from PHISICS.
+      @ In, paramDict, dictionary, dictionary of parameters
+      @ Out, None
+    """
+    with open(os.path.join(self.workingDir,'dummy.csv'), 'wb') as f:
+      instantWriter = csv.writer(f, delimiter=str(u',').encode('utf-8'),quotechar=str(u' ').encode('utf-8'), quoting=csv.QUOTE_MINIMAL)
+      instantWriter.writerow(self.joinLine(paramDict['phiDict'][1],paramDict['relDict'][1]))
+      instantWriter.writerow([0.0] * self.numOfParameters + [paramDict['relDict'][2]])
+      lineNumber = 2
+      THbetweenBurn = 0
+      mrTau = 0
+      while THbetweenBurn < len(paramDict['inpTimeDict']['TH_between_BURN'].split(' ')):
+        lineNumber = lineNumber + 1
+        if float(paramDict['relDict'][lineNumber].split(',')[paramDict['relapTimePosition']]) <= float(paramDict['inpTimeDict']['TH_between_BURN'].split(' ')[THbetweenBurn]):
+          instantWriter.writerow(self.joinLine(paramDict['phiDict'][self.timeStepSelected[mrTau] + 1],paramDict['relDict'][lineNumber]))
+        if paramDict['relDict'][lineNumber].split(',')[paramDict['relapTimePosition']] > paramDict['inpTimeDict']['TH_between_BURN'].split(' ')[THbetweenBurn]:
+          THbetweenBurn = THbetweenBurn + 1
+          mrTau = mrTau + 1
+          if THbetweenBurn == len(paramDict['inpTimeDict']['TH_between_BURN'].split(' ')):
+            instantWriter.writerow(self.joinLine(paramDict['phiDict'][paramDict['numOfPhisicsLines']],paramDict['relDict'][paramDict['numOfRelapLines']]))
+    with open(os.path.join(self.workingDir,'dummy.csv'), 'r') as infile:
```

Comment:
dummy? 
final?

Reply:
After the data are printed in the first csv dummy.csv, the values are correct but there are blank spaces and extra white space at the begining of the line. 
The files are copied from dummy to final.csv to template correctly the file. 
How would do it?